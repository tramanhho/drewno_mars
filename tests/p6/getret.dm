meow : () int { return 1; }

main : () int {
    meow();
	c : int;
	c = meow();
	give meow();
	return 0;
}