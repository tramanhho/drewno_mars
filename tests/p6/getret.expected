[BEGIN GLOBALS]

[END GLOBALS]

[BEGIN meow LOCALS]

[END meow LOCALS]

meow:    enter meow
        setret 1
        goto lbl_0
lbl_0:    leave meow

[BEGIN main LOCALS]
c-int
tmp0-int
tmp1-int
tmp2-int
[END main LOCALS]

main:    enter main

        call meow
        call meow
        getret [tmp0]
        [c] := [tmp0]
        WRITE call meow
        setret 0
        goto lbl_1
lbl_1:    leave main